validator_test.go

```diff
@@ -0,0 +1,273 @@
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.description, func(t *testing.T) {
+			assert := assert.New(t)
+			require := require.New(t)
+			msgv, err := NewTypeValidator(tc.value.m, tc.value.defaultValidators...)
+			require.NotNil(msgv)
+			require.NoError(err)
+			err = msgv.Validate(tc.value.msg)
+			if tc.expectedErr != nil {
+				assert.ErrorIs(err, tc.expectedErr)
+				return
+			}
+
+			assert.NoError(err)
+		})
+	}
+}
+
+func testNewTypeValidator(t *testing.T) {
+	type Test struct {
+		m                 map[MessageType]Validator
+		defaultValidators Validators
+	}
+
+	tests := []struct {
+		description string
+		value       Test
+		expectedErr error
+	}{
+		// Success case
+		{
+			description: "Default Validators success",
+			value: Test{
+				m: map[MessageType]Validator{
+					SimpleEventMessageType: AlwaysValid,
+				},
+				defaultValidators: Validators{AlwaysValid},
+			},
+			expectedErr: nil,
+		},
+		{
+			description: "Empty map of Validators success",
+			value: Test{
+				m:                 map[MessageType]Validator{},
+				defaultValidators: Validators{AlwaysValid},
+			},
+			expectedErr: nil,
+		},
+		{
+			description: "Omit default Validators success",
+			value: Test{
+				m: map[MessageType]Validator{
+					SimpleEventMessageType: Validators{AlwaysValid},
+				},
+			},
+			expectedErr: nil,
+		},
+		// Failure case
+		{
+			description: "Nil list of default Validators error",
+			value: Test{
+				m: map[MessageType]Validator{
+					SimpleEventMessageType: AlwaysValid,
+				},
+				defaultValidators: Validators{nil},
+			},
+			expectedErr: ErrInvalidTypeValidator,
+		},
+		{
+			description: "Empty list of Validators error",
+			value: Test{
+				m: map[MessageType]Validator{
+					SimpleEventMessageType: Validators{},
+				},
+				defaultValidators: Validators{AlwaysValid},
+			},
+			expectedErr: ErrInvalidTypeValidator,
+		},
+		{
+			description: "Nil Validator error",
+			value: Test{
+				m: map[MessageType]Validator{
+					SimpleEventMessageType: nil,
+				},
+				defaultValidators: Validators{AlwaysValid},
+			},
+			expectedErr: ErrInvalidTypeValidator,
+		},
+		{
+			description: "Nil list of Validators error",
+			value: Test{
+				m: map[MessageType]Validator{
+					SimpleEventMessageType: Validators{nil},
+				},
+				defaultValidators: Validators{AlwaysValid},
+			},
+			expectedErr: ErrInvalidTypeValidator,
+		},
+		{
+			description: "Nil map of Validators error",
+			value: Test{
+				m:                 nil,
+				defaultValidators: Validators{AlwaysValid},
+			},
+			expectedErr: ErrInvalidTypeValidator,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.description, func(t *testing.T) {
+			assert := assert.New(t)
+			msgv, err := NewTypeValidator(tc.value.m, tc.value.defaultValidators...)
+			assert.NotNil(msgv)
+			if tc.expectedErr != nil {
+				assert.ErrorIs(err, tc.expectedErr)
+				return
+			}
+
+			assert.NoError(err)
+		})
+	}
+}
+
+func testAlwaysValid(t *testing.T) {
```

Comment:
For both this and `testAlwaysInvalid`, imagine these tests as a way to prove that these functions work as expected if their logic changes in the future.  Obviously, this is unlikely.  This test provides full coverage, but not reassurance if something were to change.  My suggestion would be to add a few tests cases for both that include blatantly bad and also a good wrp.  Some examples:
- wrp with a bad MessageType
- wrp with everything filled out but some other issue - maybe a non-UTF8 string, maybe it's a Request Response but the source and destination are the same (not sure if this is considered valid or not).
- wrp with everything filled out and looking good (some examples of good wrps are already in `messages_test.go`)
- empty wrp like you have now

What you have now is good, but won't catch as many potential issues in the future if something were to change.  We don't need every possible case, just a few more for some basic sanity checking. 🙂